### ISSUE TITLE
fix fxa-auth-db-mysql startup - tries to load secret-key.json from fxa-auth-server before it exists

### ISSUE THREAD

**Issue by jrgm** [mozilla/fxa-dev#174]
I just built a new fxa-dev box from scratch, and the auth-db-server failed to start, complaining about failures to require convict. I did `supervisorctl restart fxa-auth-db-server` and it started no problem.

Looking at the timestamps on the /var/log/fxa-auth-db.log and on /data/fxa-auth-db-mysql/node_modules/*, the modules appear to have completed installing **after** supervisorctl gave up trying to get fxa-auth-db-server running. Need to look at the ordering of the tasks and handlers for this role to figure out what can be improved.

**Comment by dannycoates** [comment]
oh, weird!

**Comment by jrgm** [comment]
To clarify, the mtime of the log is only about 67 seconds behind the mtime of e.g., `node_modules/convict`. So just a small bit too early.

**Comment by jrgm** [comment]
I read the available logs but they were not giving the full story. An upcoming PR will record both stdout and stderr log messages. With that change, this message is in the stdout log.

```
{"Timestamp":1438553147090000000,"Logger":"fxa-auth-db-server",
"Type":"Error: ENOENT, no such file or directory '/data/fxa-auth-server/config/secret-key.json'",
"Severity":0,"Pid":10079,"EnvVersion":"2.0"}
```

That winds up exiting with '0', so no attempt at trying to start it again is made. However, fixing the exit code, won't fix this issue, since it will be a while before that file is created by fxa-auth-server startup after npm install. Not sure how to fix this. It does seem a bit unholy to have one service assume it can reach into another service's config files. Perhaps have fxa-dev generate keys and put them into each repos config directory?

**Comment by jrgm** [comment]
The PR in #200 "fixes" this by swapping the order of auth and auth-db roles.

**Comment by vladikoff** [comment]
blocked by Holidays :santa:
